### ISSUE TITLE
Introduce a healthCheck property to DockerCreateContainer task type

### ISSUE THREAD

**Issue by mortterna** [bmuschko/gradle-docker-plugin#754]
Docker-java library supports setting health check instruction on a created container even though image doesn't have a `HEALTHCHECK` instruction set but currently `DockerCreateContainer`task  doesn't allow any method for setting the underlying `containerCommand.withHealthcheck(...)`

https://docs.docker.com/engine/reference/builder/#healthcheck

https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/api/command/CreateContainerCmd.java#L54

**Comment by bmuschko** [comment]
Yeah, we don't have that right now. Would you be interested in providing a pull request? If you do, then we should wait until https://github.com/bmuschko/gradle-docker-plugin/pull/751 has been merged which will make it much easier for you.

**Comment by mortterna** [comment]
I don't have any gradle-plugin or groovy -fu, but I think would need something like 

```
    @Input
    @Optional
    final Property<HealthCheck> healthCheck = project.objects.property(HealthCheck)
```

...but it doesn't type-check because `docker-java-shaded` dependency seems to be quite dynamic in `DockerRemoteApiPlugin`

**Comment by bmuschko** [comment]
Definitely goes toward the right direction. Some of the fields would have to marked `@Optional` as they are not mandatory. But other than that it looks correct.

I will merge https://github.com/bmuschko/gradle-docker-plugin/pull/751 now which gets rid of the `ThreadContextClassLoader` completely. Consequently, your code would have to change as well.

We don't usually accept pull requests without a test so that would be required. It's OK if you don't have the time to work on it. It will simply take a bit of time for the committer team or the community to take on this new feature.

**Comment by bmuschko** [comment]
I am going to close this issue. We'll need someone from the community to step up and provide the functionality as a PR.

**Comment by fluxroot** [comment]
We needed support for the HEALTHCHECK command in one of our project. So I implemented PR #978. @bmuschko Could you please take a look at it and ping back if something's missing? Thanks.
